main/src/io/github/iltotore/iron/RuntimeConstraint.scala

```diff
@@ -0,0 +1,11 @@
+package io.github.iltotore.iron
+
+type RuntimeConstraint[T] = T match
+  case IronType[a, c] => RuntimeConstraint.Impl[a, c, T]
+
+object RuntimeConstraint:
+  final class Impl[A, C, T](_test: A => Boolean, val message: String):
```

Comment:
Why does it need to have a third `T` type parameter?

Reply:
This type parameter is required for opaque types so one can do something alongside what's below. It seems that the unnecessary type param was `C` in this case

```scala
object One {
  opaque type Foo = Int :| GreaterEqual[10]
  object Foo extends RefinedTypeOpsImpl[Int, GreaterEqual[10], Foo]
}

object Two {
  import One.Foo
  def doSomething(a: Int)(using RuntimeConstraint[Int, Foo]): Foo = ???
}
```

Reply:
I don't see why it should have a different shape than `Constraint[A, C]`:

```scala
final class RuntimeConstraint[A, C](_test: A => Boolean, val message: String):
  inline def test(value: A): Boolean = _test(value)
```
```scala
trait RefinedTypeOpsImpl[A, C, T](using rtcAuto: RuntimeConstraint.AutoDerived[A, C]):
  ???
```

For notes about `AutoDerived`, see my other reply below.

Reply:
My initial thinking was so that we could use it for opaque types outside of the opaque type companion object. However, thinking about it a bit more now I don't think that makes too much sense. I changed it as per your suggestion